dvc/remote/base.py

```diff
@@ -547,11 +547,49 @@ def download(
         if to_info.scheme != "local":
             raise NotImplementedError
 
-        logger.debug("Downloading '{}' to '{}'".format(from_info, to_info))
+        if self.isdir(from_info):
+            return self._download_dir(
+                from_info, to_info, name, no_progress_bar, file_mode, dir_mode
+            )
+        else:
+            return self._download_file(
+                from_info, to_info, name, no_progress_bar, file_mode, dir_mode
+            )
 
-        name = name or to_info.name
+    def _download_dir(
+        self, from_info, to_info, name, no_progress_bar, file_mode, dir_mode
+    ):
+        file_to_infos = (
+            to_info / file_to_info.relative_to(from_info)
+            for file_to_info in self.walk_files(from_info)
+        )
 
+        with ThreadPoolExecutor(max_workers=self.JOBS) as executor:
+            file_from_info = list(self.walk_files(from_info))
+            download_files = partial(
+                self._download_file,
+                name=name,
+                no_progress_bar=True,
```

Comment:
```suggestion
                no_progress_bar=no_progress_bar,
```

Reply:
This will show a bar for each file, and is much annoying when you have thousands of files.

Reply:
@verasativa I guess it is inertia from my side, as we always show progress bars for specific files in all other places, at least right now. We do plan on moving to less verbose pbars, so I agree with you that we should keep `no_progress_bar=True` here. Thanks! 🙂